frontend/src/metabase/search/components/SearchResult/SearchResult.tsx

```diff
@@ -1,126 +1,123 @@
-import { Button } from "metabase/ui";
-import { isSyncCompleted } from "metabase/lib/syncing";
-import { Icon } from "metabase/core/components/Icon";
-import Text from "metabase/components/type/Text";
+import type { LocationDescriptorObject } from "history";
+import type { MouseEvent } from "react";
+import { useCallback } from "react";
+import { push } from "react-router-redux";
 
-import { PLUGIN_MODERATION } from "metabase/plugins";
+import { useDispatch } from "metabase/lib/redux";
+import { Group, Text, Loader } from "metabase/ui";
+import { isSyncCompleted } from "metabase/lib/syncing";
 
 import type { WrappedResult } from "metabase/search/types";
-import Link from "metabase/core/components/Link";
 import { InfoText } from "../InfoText";
-import { ItemIcon, Context, Score } from "./components";
+import { ItemIcon } from "./components";
+
 import {
-  ResultButton,
-  ResultLink,
-  Title,
-  TitleWrapper,
-  Description,
-  ResultSpinner,
-  ResultLinkContent,
-  ResultInner,
+  DescriptionDivider,
+  DescriptionSection,
+  LoadingSection,
+  ModerationIcon,
+  ResultNameSection,
+  ResultTitle,
+  SearchResultContainer,
 } from "./SearchResult.styled";
 
 export function SearchResult({
   result,
   compact = false,
-  hasDescription = true,
-  onClick = undefined,
+  showDescription = true,
+  onClick = null,
   isSelected = false,
 }: {
   result: WrappedResult;
   compact?: boolean;
-  hasDescription?: boolean;
-  onClick?: (result: WrappedResult) => void;
+  showDescription?: boolean;
+  onClick?: ((result: WrappedResult) => void) | null;
```

Comment:
The initial instinct was to remove the explicit `onClick = undefined`, so that type was a quick bandaid while I wrote some other things. But this should be fine now